### PR TITLE
Use dependencies.yaml as source-of-truth for environment files.

### PR DESCRIPTION
**Pull request by cwharris**
Adds rapids-dependency-file-generator pre-commit checks for local and CI, and updates scripts to use the generated environment files.

Contributes to #1389 

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

### CODE REVIEW

File: ci/scripts/github/build.sh [hunk 1]
```diff
@@ -18,7 +18,12 @@ set -e
 
 source ${WORKSPACE}/ci/scripts/github/common.sh
 
-update_conda_env
+rapids-dependency-file-generator \
+  --output conda \
+  --file_key build \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" | tee env.yaml
```
Comment (mdemoret-nv) [inline ci/scripts/github/build.sh:24]: Should this just happen in the `update_conda_env` function?

File: ci/scripts/github/checks.sh [hunk 2]
```diff
@@ -18,7 +18,12 @@ set -e
 
 source ${WORKSPACE}/ci/scripts/github/common.sh
 
-update_conda_env
+rapids-dependency-file-generator \
+  --output conda \
+  --file_key build \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" | tee env.yaml
```
Comment (mdemoret-nv) [inline ci/scripts/github/checks.sh:24]: Should this just happen in the `update_conda_env` function?

File: ci/scripts/github/checks.sh [hunk 3]
```diff
@@ -55,9 +60,6 @@ export MORPHEUS_IN_SPHINX_BUILD=1
 rapids-logger "Checking copyright headers"
 python ${MORPHEUS_ROOT}/ci/scripts/copyright.py --verify-apache-v2 --git-diff-commits ${CHANGE_TARGET} ${GIT_COMMIT}
 
-rapids-logger "Running Python style checks"
-${MORPHEUS_ROOT}/ci/scripts/python_checks.sh
-
```
Comment (mdemoret-nv) [inline ci/scripts/github/checks.sh:60]: Were removing the python checks but are they added in pre-commit?

File: ci/scripts/github/common.sh [hunk 4]
```diff
@@ -75,31 +71,15 @@ function update_conda_env() {
...
-
-    if [[ "${SKIP_CONDA_ENV_UPDATE}" == "" ]]; then
-        rapids-logger "Checking for updates to conda env"
+    rapids-logger "Checking for updates to conda env"
 
-        # Remove default/conflicting channels from base image
+    # Remove default/conflicting channels from base image
+    if [[ -d /opt/conda/.condarc ]]; then
         rm /opt/conda/.condarc
+    fi;
```
Comment (mdemoret-nv) [inline ci/scripts/github/common.sh:79]: I dont think that this is necessary anymore and should probably be removed.

File: ci/scripts/github/docs.sh [hunk 5]
```diff
@@ -18,7 +18,12 @@ set -e
 
 source ${WORKSPACE}/ci/scripts/github/common.sh
 
-update_conda_env
+rapids-dependency-file-generator \
+  --output conda \
+  --file_key docs \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" | tee env.yaml
```
Comment (mdemoret-nv) [inline ci/scripts/github/docs.sh:24]: Should this just happen in the `update_conda_env` function?

File: ci/scripts/github/test.sh [hunk 6]
```diff
@@ -19,7 +19,12 @@ set -e
 source ${WORKSPACE}/ci/scripts/github/common.sh
 /usr/bin/nvidia-smi
 
-update_conda_env
+rapids-dependency-file-generator \
+  --output conda \
+  --file_key test \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" | tee env.yaml
```
Comment (mdemoret-nv) [inline ci/scripts/github/test.sh:25]: Should this just happen in the `update_conda_env` function?

File: dependencies.yaml [hunk 7]
```diff
@@ -21,33 +21,101 @@ files:
...
+  - huggingface
   - rapidsai
   - nvidia/label/cuda-11.8.0
   - nvidia
   - rapidsai-nightly
   - nvidia/label/dev
   - pytorch
-  - conda-forge
+  - dglteam/label/cu118
+  # - defaults
```
Comment (mdemoret-nv) [inline dependencies.yaml:102]: Does it still solve without `defaults`?

File: dependencies.yaml [hunk 8]
```diff
@@ -21,33 +21,101 @@ files:
...
 
 dependencies:
 
   build_cpp: # should be split in to `build_python` if/when converting to use scikit-build
     common:
       - output_types: [conda]
         packages:
+          - mrc=24.03 # should this be in build, or somewhere else?
```
Comment (mdemoret-nv) [inline dependencies.yaml:110]: build is good.

File: ci/scripts/github/build.sh (hunk 1 above)
Comment (cwharris) [reply ci/scripts/github/build.sh:24]: No, I think this is more the direction we want to go.

File: ci/scripts/github/checks.sh (hunk 3 above)
Comment (cwharris) [reply ci/scripts/github/checks.sh:60]: Oh good catch. I'm not intending to add the python checks in this PR.

File: ci/scripts/github/common.sh (hunk 4 above)
Comment (cwharris) [reply ci/scripts/github/common.sh:79]: This is necessary if you're running the CI container locally. It's why I added it.

File: ci/scripts/github/docs.sh (hunk 5 above)
Comment (cwharris) [reply ci/scripts/github/docs.sh:24]: No I don't think so.

File: ci/scripts/github/test.sh (hunk 6 above)
Comment (cwharris) [reply ci/scripts/github/test.sh:25]: No I don't think so. Every other RAPIDS scripts does it at this level.

File: dependencies.yaml (hunk 7 above)
Comment (cwharris) [reply dependencies.yaml:102]: It does, yes.

Comment (mdemoret-nv) [reply dependencies.yaml:119]: Ok just remove the comment then

File: ci/scripts/github/common.sh (hunk 4 above)
Comment (mdemoret-nv) [reply ci/scripts/github/common.sh:79]: Just remove the whole if block and `rm` command

**Review by mdemoret-nv — APPROVED** [review]

**Pull request merged**